pages/extensions/amp_example_preview/util/markdown_processor.py

```diff
@@ -0,0 +1,79 @@
+from grow.documents import document
+from .example_extractor import *
+
+import os
+
+EXAMPLE_TRIGGER = '[example'
+
+
+def trigger(doc, original_body, content):
+    if EXAMPLE_TRIGGER in original_body:
+        return _transform(doc, content)
+    return content
+
+
+def _transform(doc, content):
+
+    source_extractor = SourceCodeExtractor()
+    example_matches = source_extractor.find_examples_in_markdown(content)
+
+    all_imports = set()
+    all_templates = set()
+
+    pos = 0
+    count = 0
+    result = ''
+    for match in example_matches:
+
+        inline_example = match.inlineExample
+
+        count = count + 1
+
+        doc.pod.logger.debug('Found example {} in {}'.format(inline_example.index, doc.pod_path))
+
+        result = result + content[pos:match.startTagStart]
+
+        result = result + '' \
+                 + '<!-- preview id={} mode={} playground={}\n'.format(
```

Comment:
I'd prefer to encode the content into something that's easy to parse (e.g. JSON).

Reply:
Would it be possible to write these into the frontmatter and reuse grow's frontmatter parser?

Reply:
Content is json encoded now.  Because some examples contain j2 expressions to generate urls, etc we have to evaluate the source in the template. I don't think this would all work in the frontmatter, even if we could use it otherwise.